http_proxy.go

```diff
@@ -46,27 +47,18 @@ func main() {
 	}
 
 	// Middleware (runs in reverse order as they are added)
-
-	// Middleware: Forward HTTP Messages
-	forwarder, err := forward.New(nil, forward.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Handle HTTP CONNECT
-	httpConnect, err := httpconnect.New(forwarder, httpconnect.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Common request filter
-	commonHandler, err := commonfilter.New(httpConnect, testingLocal)
-	if err != nil {
-		log.Error(err)
-	}
+	filterChain := filter.NewChain(
+		commonfilter.New(&commonfilter.Options{
+			AllowLocalhost: testingLocal,
+		}),
+		httpconnect.New(&httpconnect.Options{
+			IdleTimeout: time.Duration(*idleClose) * time.Second,
+		}),
+		forward.New(&forward.Options{}),
+	)
 
 	// Create server
-	srv := server.NewServer(commonHandler)
+	srv := server.NewServer(filterChain)
```

Comment:
Note that a filter chain is just an http.Handler, so it plugs into the server just like our old starting filter used to.